project/src/store/action-creaters-middleware.ts

```diff
@@ -0,0 +1,71 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { AxiosInstance } from 'axios';
+import {
+  AuthorizationStatus,
+  ServerRoutes,
+  TIME_OUT_SHOW_ERROR,
+} from '../settings';
+import { OfferType } from '../types/offer-type';
+import { AppDispatchType, StateType } from '../types/state-type';
+import { setOffers, setAuthorizationStatus, setError, setLoadingStatus } from './action-creaters';
+import { AuthData, UserData } from '../types/user-auth-types';
+import { dropToken, saveToken } from '../services/token';
+
+type AsyncThunkType = {
+  dispatch: AppDispatchType;
+  state: StateType;
+  extra: AxiosInstance;
+};
+
+export const getOffers = createAsyncThunk<void, undefined, AsyncThunkType>(
+  'GET_OFFERS',
+  async (_args, { dispatch, extra: HTTPClient }) => {
+    const { data } = await HTTPClient.get<OfferType[]>(ServerRoutes.hotels);
+    dispatch(setLoadingStatus(true));
+    dispatch(setOffers(data));
+    dispatch(setLoadingStatus(false));
```

Comment:
```suggestion
    dispatch(setLoadingStatusTrueOffers(true));
```

это место ты не понял
надо обсудить